File: server/apps/sync-service/src/main/java/com/sharetimer/syncservice/adapter/in/listener/message/TimerUpdateTargetTimeMessage.java
package com.sharetimer.syncservice.adapter.in.listener.message;

import java.time.Instant;

public record TimerUpdateTargetTimeMessage(String timerId, String type, Payload payload) {

  public record Payload(Instant updatedAt, Instant serverTime, Instant newTargetTime) {
  }

}

File: server/libs/common/src/main/kotlin/com/sharetimer/common/message/TimerUpdateTargetTimeMessage.kt
package com.sharetimer.common.message

import java.time.Instant

data class TimerUpdateTargetTimeMessage(
    val timerId: String,
    val type: String,
    val payload: Payload,
) {
    data class Payload(
        val updatedAt: Instant,
        val serverTime: Instant,
        val newTargetTime: Instant,
    )
}
